packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/MeteringPointHostApiImpl.java

```diff
@@ -25,17 +32,34 @@ public class MeteringPointHostApiImpl implements MeteringPointHostApi {
   /** Proxy for constructor and static methods of {@link MeteringPoint}. */
   @VisibleForTesting
   public static class MeteringPointProxy {
+    Activity activity;
 
     /**
      * Creates a surface oriented {@link MeteringPoint} with the specified x, y, and size.
      *
-     * <p>A {@link SurfaceOrientedMeteringPointFactory} is used to construct the {@link
-     * MeteringPoint} because underlying the camera preview that this plugin uses is a Flutter
-     * texture that is backed by a {@link Surface} created by the Flutter Android embedding.
+     * <p>A {@link DisplayOrientedMeteringPointFactory} is used to construct the {@link
+     * MeteringPoint} because this factory handles the transformation of specified coordinates based
+     * on camera information and the device orientation automatically.
      */
     @NonNull
-    public MeteringPoint create(@NonNull Double x, @NonNull Double y, @Nullable Double size) {
-      SurfaceOrientedMeteringPointFactory factory = getSurfaceOrientedMeteringPointFactory(1f, 1f);
+    @SuppressWarnings("deprecation")
+    public MeteringPoint create(
+        @NonNull Double x,
+        @NonNull Double y,
+        @Nullable Double size,
+        @NonNull CameraInfo cameraInfo) {
+      Display display = null;
+
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        display = activity.getDisplay();
+      } else {
+        display =
+            ((WindowManager) activity.getSystemService(Context.WINDOW_SERVICE)).getDefaultDisplay();
+      }
+
+      DisplayOrientedMeteringPointFactory factory =
+          getDisplayOrientedMeteringPointFactory(display, 1f, 1f, cameraInfo);
```

Comment:
A slightly larger fix: I originally implemented this host API with a different `MeteringPointFactory`, which was not technically wrong but would have required that the plugin manually handle changes in camera and device orientation and can be tricky. This factory automatically does that based on the current camera's `CameraInfo` and the current display. So, I made changes in this PR to the `MeteringPoint` wrapper to accommodate that fact.